Calibration.py

```diff
@@ -26,44 +30,52 @@ def __init__(self, core_module, wm, oled):
 
         self.ticks = 0
 
-        # Default current mode to NONE
-        self.mode = CalibrationMode.CM_NONE
-
-    def show_motor_config(self, left):
-        """ Show motor/aux config on OLED display """
-        if self.oled is not None:
-            if left:
-                title = "Left Motor:"
-                message = str(self.core.left_servo.servo_min) + '/'\
-                    + str(self.core.left_servo.servo_mid) + '/'\
-                    + str(self.core.left_servo.servo_max)
-            else:
-                title = "Right Motor:"
-                message = str(self.core.right_servo.servo_min) + '/'\
-                    + str(self.core.right_servo.servo_mid) + '/'\
-                    + str(self.core.right_servo.servo_max)
-
-            self.oled.cls()  # Clear Screen
-            self.oled.canvas.text((10, 10), title, fill=1)
-            self.oled.canvas.text((10, 30), message, fill=1)
-            # Now show the mesasge on the screen
-            self.oled.display()
+        # Note: string desciptions on the right are for
+        # reference only. Not currently used in this code.
+        self.menu_list = OrderedDict((
+            (ServoEnum.LEFT_MOTOR_ESC, "Left Motor ESC"),
+            (ServoEnum.RIGHT_MOTOR_ESC, "Right Motor ESC"),
+            (ServoEnum.LEFT_AUX_ESC, "Left Aux ESC"),
+            (ServoEnum.RIGHT_AUX_ESC, "Right Aux ESC"),
+            (ServoEnum.LEFT_FLINGER_ESC, "Left Flinger ESC"),
+            (ServoEnum.RIGHT_FLINGER_ESC, "Right Flinger ESC"),
+            (ServoEnum.LEFT_FLINGER_SERVO, "Left Flinger Servo"),
+            (ServoEnum.RIGHT_FLINGER_SERVO, "Right flinger Servo"),
+            (ServoEnum.WINCH_SERVO, "Winch Servo")
+        ))
 
-    def show_aux_1_config(self, left):
-        """ Show motor/aux config on OLED display """
+        # Default current mode to NONE
+        self.mode = ServoEnum.LEFT_MOTOR_ESC
+
+    def get_next_mode(self, mode):
+        """ Find the previous menu item """
+        mode_index = self.menu_list.keys().index(mode)
+        next_index = mode_index + 1
+        if next_index >= len(self.menu_list):
+            next_index = 0  # Wrapped round to end
+        return self.menu_list.keys()[next_index]
+
+    def get_previous_mode(self, mode):
+        """ Find the previous menu item """
+        mode_index = self.menu_list.keys().index(mode)
+        previous_index = mode_index - 1
+        if previous_index < 0:
+            previous_index = len(self.menu_list) - 1  # Wrapped round to end
```

Comment:
you're still thinking in C++, negative indexing is totally legit in python. ;-)

Just wait until you bump in to slicing notation. The optional step paramter will blow your mind:

```
l = [1, 2, 3, 4, 5]
l[1] => 2
l[1:] => [2, 3, 4, 5]
l[:1] => [1]
l[::2] => [1, 3, 5]
l[::-2] => [5, 3, 1]
```
TL;DR: Python is *awesome* with lists ;-)